forum/frontend/src/context/AuthContext.js

```diff
@@ -0,0 +1,35 @@
+import React, { createContext, useState, useEffect } from 'react';
+
+export const AuthContext = createContext();
+
+export const AuthProvider = ({ children }) => {
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  const [role, setRole] = useState('unassigned');
+
+  useEffect(() => {
+    const token = localStorage.getItem('accessToken');
+    setIsAuthenticated(!!token);
+  }, []);
+
+  const login = (token) => {
+    localStorage.setItem('accessToken', token);
+    setIsAuthenticated(true);
+  };
+
+  const logout = () => {
+    localStorage.removeItem('accessToken');
+    setIsAuthenticated(false);
+    setRole('unassigned');
+  };
+
+  const changeRole = (newRole) => {
+    setRole(newRole);
+    localStorage.setItem('role', newRole);
+  };
+
+  return (
+    <AuthContext.Provider value={{ isAuthenticated, role, login, logout, changeRole }}>
+      {children}
+    </AuthContext.Provider>
+  );
+};
```

Comment:
Storing the token in `localStorage` does not account for token expiration, and there is no automatic logout when the token becomes invalid. Use JWT expiration handling with a periodic check to refresh or invalidate the token.